crates/wasi-cloud/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::{Error, Result};
+use reqwest::Client;
+use spin_common::table::Table;
+use spin_core::HostComponent;
+use std::sync::Arc;
+use tokio::sync::Notify;
+use wit::types2 as types;
+
+pub mod http;
+mod poll;
+mod streams;
+
+pub mod wit {
+    wasmtime::component::bindgen!({
+        path: "../../wit/preview2",
+        world: "proxy",
+        async: true
+    });
+}
+
+pub struct WasiCloudComponent;
+
+impl HostComponent for WasiCloudComponent {
```

Comment:
Yes, I was thinking we'd add all the `wasi-cloud-core` state to this component (including `wasi-keyvalue`), partly because I expect they'll need to share the same `poll` and `streams` tables.  I'm open to other ways of organizing this, though.

Reply:
I think we'll want to bake some of these common preview2 types into `spin-core`. I'm expecting to see them in `wasmtime-wasi` 10 anyway.